### ISSUE TITLE
Possible regression in CVC4

### ISSUE THREAD

**Issue by marcogario** [pysmt/pysmt#205]
With the latest version of CVC4 (b455f5cde8b84b7951d309604b75a76afd8b8bfa) tests on SMT-LIB are going into timeout. This is an example:

```
$ time python shell.py --file pysmt/test/smtlib/small_set/QF_LIA/prp-23-47.smt2 
CVC4: Calling Check-SAT
unsat

real	0m31.337s
user	0m31.236s
sys	0m0.088s

$ time .smt_solvers/CVC4_bin/bin/cvc4 pysmt/test/smtlib/small_set/QF_LIA/prp-23-47.smt2 
unsat
real	0m0.516s
user	0m0.496s
sys	0m0.020s
```

The parsing and conversion towards CVC4 is fast. The time is spent in the solving. Unfortunately, I cannot reproduce the problem when going through an SMT-LIB file.

I think this might have to do with new defaults for CVC4.

**Comment by marcogario** [comment]
As a side remark, we should compile CVC with ''--best --enable-gpl"

**Comment by marcogario** [comment]
I tested with the current configuration, and the problem is present also there. Therefore.

- I will proceed to upgrade CVC4 to the latest version
- We need to better understand the options that are set by default from the command-line (i.e., the example above seems to use ITE-Simplification, that is disabled by default).

**Comment by marcogario** [comment]
I identified the regression has been introduced in https://github.com/CVC4/CVC4/commit/d101e7fed051685673c13317cb45166ba5ef7798 .

During incremental solving (I guess API calls follow this rule) the ITE simplification is disabled. 
Maybe there is a way to re-enable this from configuration.

**Comment by mikand** [comment]
This is related to #233 . We should update CVC4 version as sson as possible.

**Comment by marcogario** [comment]
This seems to be fixed with #250 . In #251 we upgrade to the latest CVC4 version.
